ci/scripts/cpp_test.sh

```diff
@@ -114,6 +120,21 @@ else
     "$@"
 fi
 
+if [ "${ARROW_USE_MESON:-OFF}" = "OFF" ] && [ "${ARROW_EMSCRIPTEN:-OFF}" = "OFF" ]; then
+  CMAKE_PREFIX_PATH="${CMAKE_INSTALL_PREFIX:-${ARROW_HOME}}"
+  if [ -n "${VCPKG_ROOT}" ] && [ -n "${VCPKG_TARGET_TRIPLET}" ]; then
+    CMAKE_PREFIX_PATH+=";${VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}"
+  fi
+  cmake \
+    -S ${source_dir}/examples/minimal_build \
+    -B ${build_dir}/examples/minimal_build \
+    -DCMAKE_PREFIX_PATH="${CMAKE_PREFIX_PATH}"
+  cmake --build ${build_dir}/examples/minimal_build
+  pushd ${source_dir}/examples/minimal_build
+  ${build_dir}/examples/minimal_build/arrow-example
+  popd
+fi
```

Comment:
This is a test for our CMake packages. If `find_package(Arrow)` doesn't work, this will detect it.

JNI build uses `find_package(Arrow)`.